noir_stdlib/src/hash/poseidon/bn254/perm.nr

```diff
@@ -0,0 +1,291 @@
+                       consts::mds8(),
+                       state);
+
+    for i in 0..8
+    {
+        out[i] = perm[i];
+    }
+
+    out
+}
+
+fn x5_9(state: [Field; 9]) -> [Field; 9]
+{
+    let mut out = state;
+    let perm = permute(consts::rf,
+                       consts::rp()[7],
+                       consts::alpha,
+                       consts::ark9(),
+                       consts::mds9(),
+                       state);
+
+    for i in 0..9
+    {
+        out[i] = perm[i];
+    }
+
+    out
+}
+
+fn x5_10(state: [Field; 10]) -> [Field; 10]
+{
+    let mut out = state;
+    let perm = permute(consts::rf,
+                       consts::rp()[8],
+                       consts::alpha,
+                       consts::ark10(),
+                       consts::mds10(),
+                       state);
+
+    for i in 0..10
+    {
+        out[i] = perm[i];
+    }
+
+    out
+}
+
+fn x5_11(state: [Field; 11]) -> [Field; 11]
+{
+    let mut out = state;
+    let perm = permute(consts::rf,
+                       consts::rp()[9],
+                       consts::alpha,
+                       consts::ark11(),
+                       consts::mds11(),
+                       state);
+
+    for i in 0..11
+    {
+        out[i] = perm[i];
+    }
+
+    out
+}
+
+fn x5_12(state: [Field; 12]) -> [Field; 12]
+{
+    let mut out = state;
+    let perm = permute(consts::rf,
+                       consts::rp()[10],
+                       consts::alpha,
+                       consts::ark12(),
+                       consts::mds12(),
+                       state);
+
+    for i in 0..12
+    {
+        out[i] = perm[i];
+    }
+
+    out
+}
+
+fn x5_13(state: [Field; 13]) -> [Field; 13]
+{
+    let mut out = state;
+    let perm = permute(consts::rf,
+                       consts::rp()[11],
+                       consts::alpha,
+                       consts::ark13(),
+                       consts::mds13(),
+                       state);
+
+    for i in 0..13
+    {
+        out[i] = perm[i];
+    }
+
+    out
+}
+
+fn x5_14(state: [Field; 14]) -> [Field; 14]
+{
+    let mut out = state;
+    let perm = permute(consts::rf,
+                       consts::rp()[12],
+                       consts::alpha,
+                       consts::ark14(),
+                       consts::mds14(),
+                       state);
+
+    for i in 0..14
+    {
+        out[i] = perm[i];
+    }
+
+    out
+}
+
+fn x5_15(state: [Field; 15]) -> [Field; 15]
+{
+    let mut out = state;
+    let perm = permute(consts::rf,
+                       consts::rp()[13],
+                       consts::alpha,
+                       consts::ark15(),
+                       consts::mds15(),
+                       state);
+
+    for i in 0..15
+    {
+        out[i] = perm[i];
+    }
+
+    out
+}
+
+fn x5_16(state: [Field; 16]) -> [Field; 16]
+{
+    let mut out = state;
+    let perm = permute(consts::rf,
+                       consts::rp()[14],
```

Comment:
Function calls should also either be all on one line or with each arg (including the first) on a newline with standard indenting.